Framework/PythonInterface/plugins/algorithms/SaveNexusPD.py

```diff
@@ -150,7 +152,7 @@ def _writeX(self, nxdata, name, wksp, index, writeDx):
             temp.attrs['units'] = units
 
     def _writeProtonCharge(self, nxentry, wksp):
-        if not 'gd_prtn_chrg' in wksp.run().keys():
+        if not 'gd_prtn_chrg' in list(wksp.run().keys()):
```

Comment:
Yes, `Run` defines `dict`-like methods so `if not 'gd_prtn_chrg' in wksp.run()` should work